modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessorBuilder.java

```diff
@@ -0,0 +1,118 @@
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.runtime.operation;
+
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import static org.mule.runtime.api.util.Preconditions.checkArgument;
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.api.meta.model.ComponentModel;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.config.ConfigurationException;
+import org.mule.runtime.core.api.registry.RegistrationException;
+import org.mule.runtime.core.api.retry.policy.RetryPolicyTemplate;
+import org.mule.runtime.core.api.streaming.CursorProviderFactory;
+import org.mule.runtime.core.internal.policy.PolicyManager;
+import org.mule.runtime.extension.api.runtime.config.ConfigurationProvider;
+import org.mule.runtime.module.extension.internal.runtime.ExtensionComponent;
+import org.mule.runtime.module.extension.internal.runtime.connectivity.ExtensionConnectionSupplier;
+import org.mule.runtime.module.extension.internal.runtime.connectivity.oauth.ExtensionsOAuthManager;
+import org.mule.runtime.module.extension.internal.runtime.resolver.ParametersResolver;
+import org.mule.runtime.module.extension.internal.runtime.resolver.ResolverSet;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class ComponentMessageProcessorBuilder<M extends ComponentModel, P extends ExtensionComponent> {
+
+  protected final ExtensionModel extensionModel;
+  protected final M operationModel;
+  protected final PolicyManager policyManager;
+  protected final MuleContext muleContext;
+  protected final ExtensionConnectionSupplier extensionConnectionSupplier;
+  protected final ExtensionsOAuthManager oauthManager;
+
+  protected ConfigurationProvider configurationProvider;
+  protected Map<String, ?> parameters;
+  protected String target;
+  protected String targetValue;
+  protected CursorProviderFactory cursorProviderFactory;
+  protected RetryPolicyTemplate retryPolicyTemplate;
+
+  public ComponentMessageProcessorBuilder(ExtensionModel extensionModel,
+                                          M operationModel,
+                                          PolicyManager policyManager,
+                                          MuleContext muleContext) {
+
+    checkArgument(extensionModel != null, "ExtensionModel cannot be null");
+    checkArgument(operationModel != null, "OperationModel cannot be null");
+    checkArgument(policyManager != null, "PolicyManager cannot be null");
+    checkArgument(muleContext != null, "muleContext cannot be null");
+
+    this.extensionModel = extensionModel;
+    this.operationModel = operationModel;
+    this.policyManager = policyManager;
+    this.muleContext = muleContext;
+    extensionConnectionSupplier = lookup(ExtensionConnectionSupplier.class);
```

Comment:
why lookup? If you can't `@Inject` for whatever reason, I'd rather have this passed in the constructor

Reply:
no idea, it's a copy/paste from the message processor TBH

Reply:
did you delete OperationMessageProcessor? How is this class any different?